osu.Game.Rulesets.Osu/Difficulty/OsuPerformanceCalculator.cs

```diff
@@ -31,8 +31,8 @@ public class OsuPerformanceCalculator : PerformanceCalculator
         private int countMeh;
         private int countMiss;
 
-        public OsuPerformanceCalculator(Ruleset ruleset, WorkingBeatmap beatmap, ScoreInfo score)
-            : base(ruleset, beatmap, score)
+        public OsuPerformanceCalculator(Ruleset ruleset, WorkingBeatmap beatmap, ScoreInfo score, DifficultyAttributes attributes)
+            : base(ruleset, beatmap, score, attributes)
         {
             countHitCircles = Beatmap.HitObjects.Count(h => h is HitCircle);
```

Comment:
The `WorkingBeatmap` is only ever used for this in any of the performance calculators, right? I'm quite concerned that retrieving the playable beatmap will hit performance very hard, and this is done for every score that's opened on the results screen.

Unsure what the path forward is here yet, just putting this out there as something that needs investigation to make sure we can actually handle this sort of load. This is similar to the reason `BeatmapDifficultyManager` uses a task scheduler.

Reply:
Actually, do we want to keep a `WorkingBeatmap` or even a `IBeatmap` arround for performance calculation ? Out of the 4 rulesets, only std seems to use the `IBeatmap` in `PerformanceCalculator` only to get the number of sliders and hit circles. Can't these calculations be moved inside the difficulty attributes calculation?

 Wouldn't doing so make the calculation lighter performance-wise since we're assured to have difficulty attributes cached in `BeatmapDifficultyManager` and we don't have the burden of loading the working beatmap ? This way we could just construct the performance calculator with a null working beatmap and the cached difficulty attributes fetched from`BeatmapDifficultyManager`

Reply:
If we do decide that performance calculation should be based entirely on difficulty attributes then `PerformanceCalculator` should have nothing to do with any `WorkingBeatmap` on its own. (Difficulty calculation is a different thing entirely.)

Reply:
Yeah, I agree with completely removing the beatmap from performance calculator, and heaving the count as part of the attributes.